udata/core/dataset/search.py

```diff
@@ -169,22 +169,25 @@ def is_indexable(cls, dataset):
 
     @classmethod
     def serialize(cls, dataset):
-        org_id = (str(dataset.organization.id)
-                  if dataset.organization is not None else None)
         if dataset.organization:
-            image_url = dataset.organization.logo(40, external=True)
+            organization = Organization.objects(id=dataset.organization.id).first()
+            image_url = organization.logo(40, external=True)
+            owner = None
         elif dataset.owner:
-            image_url = dataset.owner.avatar(40, external=True)
+            owner = User.objects(id=dataset.owner.id).first()
+            image_url = owner.avatar(40, external=True)
+            organization = None
         else:
+            organization = None
```

Comment:
Maybe initialize `organization`, `owner` and `image_url` to `None` before the `ifs`? If feel more comfortable with this pattern but YMMV.

Reply:
You're right, it's much better